### DISCUSSION TITLE
ConnectKit should provide a ui-handled SIWE trigger

### DISCUSSION THREAD

**Discussion by shrugs** [family/connectkit#146] [Ideas]
i'm implementing SIWE w/ connectkit and was overall very pleased with the integration, but a minor behavior that's bothering me is that CK provides a `show` handler for the connectbutton that opens the CK ui, but the provided `signIn` handler just executes SIWE logic, without UI. it would be vastly preferrable, imo, if there were a handler that opened the modal to the SIWE route and (also) triggered the `signIn` flow so that in the situation where the modal isn't already open (either auth has expired or the user connected and dismissed the modal) we get CK's nice UI

**Comment by lochie** [comment]
Great suggestion! Have started a PR https://github.com/family/connectkit/pull/152 to add this functionality to our `useModal` Hook

Thinking this would look something like this:
```tsx
const { openSIWE } = useModal();

<button onClick={() => openSIWE()}>Open to SIWE page</button>
<button onClick={() => openSIWE(true)}>Open to SIWE page and prompt Sign In With Ethereum</button>
```
